examples/tile-layer-directive/index.ts

```diff
@@ -2,88 +2,138 @@
+      [detectRetina]="getInputPropertyByName('detectRetina').value"
+      [crossOrigin]="getInputPropertyByName('crossOrigin').value"
+      ></yaga-tile-layer>
     </yaga-map>
   </div>
-  
-  <div class="starter-template">
-    <h1>Layertree</h1>
-    <ol>
-  <li *ngFor="let layer of tileLayers">
-    <input class="form-control" type="text" [(ngModel)]="layer.url">
-    <input class="form-control" type="range" max="1" min="0" step="0.05" [(ngModel)]="layer.opacity">
-    <button class="btn btn-danger" (click)="removeLayer(layer)">Remove</button>
-  </li>
-  <li>
-    <input type="text" [(ngModel)]="newLayerUrl">
-    <input type="range" max="1" min="0" step="0.05" [(ngModel)]="newLayerOpacity">
-    <button (click)="addLayer()">Add</button>
-  </li>
-</ol>
-    </div>
-  
+  ${ PROPERTIES_WRAPPER }
+  <div>
+  </div>
 </div><!-- /.container -->
+<example-footer></example-footer>
 
 `;
 /* tslint:enable */
 
-export interface ITileLayerOptions {
-    url: string;
-    opacity?: number;
-}
-
 @Component({
     selector: 'app',
     template
 })
-export class AppComponent implements AfterViewInit {
-    public zoom: number = 10;
-    public lat: number = 51;
-    public lng: number = 7;
-    public newLayerUrl: string = 'https://b.tile.opentopomap.org/{z}/{x}/{y}.png';
-    public newLayerOpacity: number = 1;
-
-    public tileLayers: ITileLayerOptions[] = [{url: 'http://b.tile.openstreetmap.org/{z}/{x}/{y}.png', opacity: 1}];
-
-    @ViewChild(MapComponent) private mapComponent: MapComponent;
-
-    constructor() {
-        (<any>window).app = this;
-    }
-    public removeLayer(layer: ITileLayerOptions): void {
-        this.tileLayers.splice(this.tileLayers.indexOf(layer), 1);
-    }
-    public addLayer(): void {
-        this.tileLayers.push({url: this.newLayerUrl, opacity: this.newLayerOpacity});
-    }
+export class AppComponent extends ExampleAppComponentBlueprint {
+    public properties: IExampleProperties = {
+        duplex: [
+            {name: 'url', value: OSM_TILE_LAYER_URL, type: 'text' },
+            {name: 'display', value: true, type: 'checkbox' },
+            {name: 'opacity', value: 7, type: 'relative' },
+            {name: 'zIndex', value: 5, type: 'number' }
+        ],
+        input: [
+            // {name: 'tileSize', value: true, type: 'point' },
+            {name: 'updateWhenIdle', value: true, type: 'checkbox' },
+            {name: 'updateWhenZooming', value: true, type: 'checkbox' },
+            {name: 'updateInterval', value: true, type: 'checkbox' },
+            // {name: 'bounds', value: true, type: 'bounds' },
+            {name: 'noWrap', value: true, type: 'checkbox' },
+            {name: 'className', value: '', type: 'text' },
+            {name: 'keepBuffer', value: true, type: 'checkbox' },
+            {name: 'maxNativeZoom', value: 15, type: 'number' },
+            {name: 'minNativeZoom', value: 5, type: 'number' },
+            // {name: 'subdomains', value: true, type: 'array-of-string' },
```

Comment:
The `array-of-string` type we also need in a lot of other directives ... e.g. ... WMS, polyline, polygon, attributuion, ..., ..., ...